### ISSUE TITLE
Enhancement: Add unit tests for preprocessing module (chunker.py, cleaners.py, curate_benchmark.py)

### ISSUE THREAD

**Issue by Tejasri-T** [SidakSethi-Singh/LawSathi-Rag#43]
## Description

The `tests/` directory currently contains only 2 test files covering evaluation and benchmark corpus deduplication:

* `tests/test_error_analysis.py` — 8 tests for error classification
* `tests/test_run_full_benchmark.py` — 3 tests for corpus deduplication

The entire preprocessing module currently has **zero test coverage**:

| File                                    | Functions                                                                                           | Tests  |
| --------------------------------------- | --------------------------------------------------------------------------------------------------- | ------ |
| `src/preprocessing/chunker.py`          | `chunk_text()`, `chunk_documents()`                                                                 | ❌ None |
| `src/preprocessing/cleaners.py`         | `strip_html_tags()`, `normalize_spacing()`, `ensure_utf8_encoding()`, `clean_text()`                | ❌ None |
| `src/preprocessing/curate_benchmark.py` | `find_data_file()`, `load_raw_data()`, `extract_qa_fields()`, `process_record()`, `split_records()` | ❌ None |

These modules handle data integrity for the entire benchmark. Any regression here could silently corrupt the evaluation dataset.

### `tests/test_chunker.py`

* Empty string input → returns `[]`
* Text shorter than `chunk_size` → returns a single chunk
* Text exactly at `chunk_size` boundary → no off-by-one errors
* Overlap parameter works correctly → chunks share the expected tokens
* Large document produces multiple chunks
* `chunk_documents()` flattens results from multiple documents

### `tests/test_cleaners.py`

* `strip_html_tags("<p>Hello</p>")` → `"Hello"`
* `normalize_spacing("hello   world\n\n")` → `"hello world"`
* `ensure_utf8_encoding()` handles non-UTF-8 characters
* `clean_text()` applies the full cleaning pipeline in the correct order
* Empty string and `None`-like inputs are handled gracefully

### `tests/test_curate_benchmark.py`

* `extract_qa_fields()` with standard keys (`question`, `answer`, `context`)
* `extract_qa_fields()` with alternate keys (`query`, `response`, `passage`)
* `extract_qa_fields()` raises `ValueError` for unknown schemas
* `split_records()` produces the correct 70/15/15 split ratios
* `split_records()` caps the test set at 200 records
* `split_records()` moves excess test records to the training set

## Acceptance Criteria

* [ ] `tests/test_chunker.py` created with **≥ 5 test cases**
* [ ] `tests/test_cleaners.py` created with **≥ 5 test cases**
* [ ] `tests/test_curate_benchmark.py` created with **≥ 5 test cases**
* [ ] All tests pass using:

```bash
python -m pytest tests/ -v
```

* [ ] No external API calls or file system side effects occur during tests
* [ ] Mocks and fixtures are used where neces

**Comment by ShruthiNarasimhan** [comment]
Hi @Tejasri-T, I'd like to work on this issue as part of OSCI'26.

I went through the preprocessing modules and the existing test coverage. I can add isolated pytest coverage for chunker.py, cleaners.py, and curate_benchmark.py, including the boundary cases and alternate input schemas mentioned in the issue, while keeping the tests free of external API calls and filesystem side effects.

I'll also make sure the full tests/ suite passes after adding the new test coverage.

I'd be happy to take this up and submit a PR.

**Comment by Garcia-786** [comment]
Hi @SidakSethi-Singh, I'd love to take this up as part of OSCI'26.

I can add isolated pytest coverage for chunker.py, cleaners.py, and curate_benchmark.py, implementing the suggested test cases and boundary checks while keeping tests completely free of external API calls and filesystem side effects. I'll also ensure the full tests/ suite passes cleanly.
